docs/walkthroughs/make_fossa_deps_conan.py

```diff
@@ -0,0 +1,184 @@
+                fossa_deps_yaml.append(f"- name: {json.dumps(dep.name)}")
+                fossa_deps_yaml.append(f"  version: {json.dumps(dep.version)}")
+                fossa_deps_yaml.append(f"  path: {json.dumps(dep.path)}")
+                fossa_deps_yaml.append("\n")
+
+        if self.custom_dependencies:
+            fossa_deps_yaml.append("custom-dependencies:")
+            for dep in self.custom_dependencies:
+                fossa_deps_yaml.append(f"- name: {json.dumps(dep.name)}")
+                fossa_deps_yaml.append(f"  version: {json.dumps(dep.version)}")
+                fossa_deps_yaml.append(f"  license: {json.dumps(dep.license)}")
+
+                if dep.metadata is not None and (dep.metadata.homepage is not None or dep.metadata.description is not None):
+                    fossa_deps_yaml.append("  meatdata:")
+                    if dep.metadata.homepage:
+                        fossa_deps_yaml.append(f"    homepage: {json.dumps(dep.metadata.homepage)}")
+                    if dep.metadata.description:
+                        fossa_deps_yaml.append(f"    description: {json.dumps(dep.metadata.description)}")
+                fossa_deps_yaml.append("\n")
+
+        with open('fossa-deps.yml', 'w+') as f:
+            f.writelines(s + '\n' for s in comments + fossa_deps_yaml)
+
+
+def name_version_of(label: str) -> Tuple[str, str]:
+    if "/" not in label:
+        raise ValueError(f"coult not parse name and version for: {label}")
+    name, version = label.split("/", 1)
+    return name, version
+
+def license_of(node: dict) -> Optional[str]:
+    return node.get("license")
+
+def homepage_of(node: dict) -> Optional[str]:
+    candidate = node.get("homepage")
+    if not candidate:
+        return node.get("url")
+    return candidate
+
+def description_of(node: dict) -> Optional[str]:
+    return node.get("description")
+
+def mk_fossa_deps(graph):
+    if 'root' not in graph or 'nodes' not in graph:
+        raise ValueError("root and nodes must exist in the conan graph")
+
+    vendored_deps = []
+    custom_deps = []
+    for node in graph.get('nodes', []):
+        label = node.get("label")
+        if label.lower() in ["conanfile.txt", "conanfile.py"]:
+            logging.info(f"excluding {label} from fossa-deps, as this is a manifest file, not a dependency")
+            continue
+
+        if node.get("test", False):
+            continue
+
+        # https://docs.conan.io/2.0/tutorial/consuming_packages/cross_building_with_conan.html
+        if "build" == node.get("context"):
+            logging.info(f"excluding {label} from fossa-deps, as this package as build context, and is build dependency.")
+            continue
+
+        pkg_id = node.get("package_id", "none")
+        name, raw_version = name_version_of(label)
+        version_params = urllib.parse.urlencode({'package_id': pkg_id}, doseq=True)
+        version = f"{raw_version},{version_params}"
```

Comment:
I chose to go with `,` as separator for vendor dependency instead of `#`, or url query parameter, as there is defect, which causes archive upload to fail, if `?` or `#` are used in revision value. This should not be as much of concern since this is doing archive upload, and no locator revision spec format is relevant for other FOSSA core functionalities.